servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/LoadBalancingPolicy.java

```diff
@@ -0,0 +1,12 @@
+package io.servicetalk.loadbalancer;
+
+/**
+ * Definition of the selector mechanism used for load balancing.
+ */
+public interface LoadBalancingPolicy {
```

Comment:
+1 for going forward with this approach, this is more aligned with other use-cases, like configuring different HTTP protocol versions on client/server builder and makes it easier to manage externally via ServiceLoader providers if there is only one LB builder with only one provider. 

Not now, but in the future we may need to add a "selector" as a method on `LoadBalancingPolicy` to let users implement their own selectors. For now, +1 for keeping that detail pkg-private.